templates/managed-rds-postgresql-v1.j2

```diff
@@ -0,0 +1,284 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: >-
+  Provision RDS PostgreSQL instance
+Parameters:
+  VpcSubnet:
+    Type: List<AWS::EC2::Subnet::Id>
+    ConstraintDescription: must be an existing subnet
+    Default: PrivateSubnet
+    AllowedValues:
+      - PrivateSubnet
+      - PrivateSubnet1
+      - PrivateSubnet2
+      - PublicSubnet
+      - PublicSubnet1
+      - PublicSubnet2
+  KmsKeyId:
+    Description: The Amazon Resource Name (ARN) of the AWS Key Management Service master key that is used to create the encrypted volume
+    Type: String
+    Default: ""
+  DBPortNumber:
+    Description: The port number on which the database accepts connections.
+    Type: Number
+    Default: '5432'
+    MinValue: '1150'
+    MaxValue: '65535'
+    ConstraintDescription: Valid values range from 1150-65535
+  DBMasterUsername:
+    Description: The database admin account username
+    Type: String
+    MinLength: '5'
+    MaxLength: '16'
+    AllowedPattern: ^[\w]+$
+    ConstraintDescription: Minimum 5 characters must begin with a letter and contain
+      only alphanumeric and underscore characters.
+  DBMasterUserPassword:
+    Description: The password for the master user. The password can include any printable
+      ASCII character except "/", "\"", or "@".
+    NoEcho: true
+    Type: String
+    MinLength: '12'
+    MaxLength: '41'
+    AllowedPattern: ^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?\d)(?=.*?(_|\W))(?=.*?\s)?.{8,}$
+    ConstraintDescription: 'Minimum 12 characters, at least one of each of the following:
+      uppercase, lowercase, number, and symbol character such as !#$%^&*()<>[]{}|_+-=.'
+  DBBackupRetentionPeriod:
+    Description: The number of days for which automated backups are retained. Allowed
+      values 0-10. Setting this parameter to a positive number enables backups. Setting
+      this parameter to 0 disables automated backups.
+    Default: '1'
+    Type: Number
+    MinValue: '0'
+    MaxValue: '10'
+    ConstraintDescription: must be a number between 0 and 10.
+  VolumeSize:
+    Default: '20'
+    Description: General Purpose (SSD) storage (gp2). Must be an integer from 20 to
+      16384.
+    Type: Number
+    MinValue: '20'
+    MaxValue: '16384'
+    ConstraintDescription: Must be an integer from 20 to 16384.
+  EncryptVolume:
+    Default: 'true'
+    Description: Local encryption on the instance
+    Type: Boolean
+  DBEngineVersion:
+    Description: The database engine version
+    Type: String
+    Default: '11.2'
+    AllowedValues:
+      - '11.5R1'
+      - '11.2'
+      - '11.1'
+      - '10.7'
+      - '10.6'
+      - '10.5'
+      - '10.4'
+      - '10.3'
+      - '10.1'
+  InstanceType:
+    Description: The database instance type
+    Type: String
+    Default: db.t2.small
+    AllowedValues:
```

Comment:
are all these instance types really possible?  https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.DBInstanceClass.html#Concepts.DBInstanceClass.Types

Reply:
yah, they should all be possible but I paired the list back to just a few reasonable ones.

Reply:
Makes sense to me. Can that field take wildcards? I limited it further than that, allowing only larger versions of the optimized instance types since there's really no reason to have a small, memory-optimized instance for our cases.